core/src/com/unciv/ui/utils/ExtensionFunctions.kt

```diff
@@ -284,3 +309,36 @@ object UncivDateFormat {
      */
     fun String.parseDate(): Date = utcFormat.parse(this)
 }
+
+
+/**
+ * Returns a wrapped version of a function that safely crashes the game to [CrashScreen] if an exception or error is thrown.
+ *
+ * In case an exception or error is thrown, the return will be null. Therefore the return type is always nullable.
+ *
+ * @param postToMainThread Whether the [CrashScreen] should be opened by posting a runnable to the main thread, instead of directly. Set this to true if the function is going to run on any thread other than the main loop.
+ * @return Result from the function, or null if an exception is thrown.
+ * */
+fun <R> (() -> R).wrapCrashHandling(postToMainThread: Boolean = false): () -> R? = {
+        try {
+            this()
+        } catch (e: Throwable) {
+            if (postToMainThread) {
+                Gdx.app.postRunnable {
+                    UncivGame.Current.setScreen(CrashScreen(e))
+                }
+            } else {
+                UncivGame.Current.setScreen(CrashScreen(e))
+            }
+            null
+        }
+    }
```

Comment:
Why not always post to main thread? That way you don't need to worry about whether you're running from the main thread or not

Reply:
`postRunnable` goes through a different call path than calling directly. Since this is meant to be done after a crash, where the application state is undefined, I'm trying to play things safe by avoiding different lower-level behaviour where possible.

E.G.:
- `postRunnable`, I think, means that whatever called the wrapped function will get and process its returned result *before* the crash is later handled in the next game loop. In such a case, the return result is always null, so the behaviour after the return during this game loop is kinda undefined and in theory I think it's safer to handle the crash right away when possible than wait until the next frame.
- Usually everything on the main thread goes through either `Stage.[touchUp/touchDown/someEvent]()` or `Game.render()`. The stuff given to `postRunnable` skips that. I don't know of any specific problems that would cause, but why not play it safe by minimizing the number of different GDX API paths we're using when the game is already in an undefined state?

Reply:
That said, having taken a closer at the stack, I think the benefit might be marginal if any… With LWJGL3, at least, the calls for `postRunnable` are a subset of those used for events and rendering, diverging at `Application.loop()` and jumping straight into the lambda.

LMK If I should always post, given that.